### PR TITLE
Provide more user data in embed

### PR DESCRIPTION
**Pull request by asiia-trilitech**
## Proposed changes

[Task link](https://app.asana.com/0/1207646563234653/1207749664422716/f)

[Kukai way of handling data](https://github.com/kukai-wallet/kukai/blob/3ebc2bc040edbaf7783fd4224c098223e6af4d75/src/app/services/torus/torus.service.ts#L215)

Depends on [umami-embed change](https://github.com/trilitech/umami-embed-npm/pull/10)

#### Google
- emailAddress: `email.coogle.com`
- id: `email.coogle.com`
- imageUrl: `google avatar url`
- name: `Name Surname`
- typeOfLogin: `google`

#### Reddit
- emailAddress: `undefined`
- id: `reddit username`
- imageUrl: `reddit avatar`
- name: `reddit username`
- typeOfLogin: `reddit`

#### Twitter
- emailAddress: `undefined`
- id: `twitter id`
- imageUrl: `twitter avatar`
- name: `Name Surname`
- typeOfLogin: `twitter`

#### Facebook
- emailAddress: `test@email.com`
- id: `facebook id`
- imageUrl: `facebook avatar`
- name: `Name Surname`
- typeOfLogin: `facebook`

## Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor
- [ ] Breaking change
- [ ] UI fix

## Checklist

- [ ] Tests that prove my fix is effective or that my feature works have been added
- [ ] Documentation has been added (if appropriate)
- [ ] Screenshots are added (if any UI changes have been made)
- [ ] All TODOs have a corresponding task created (and the link is attached to it)

### CODE REVIEW

File: packages/social-auth/src/Auth.ts [hunk 1]
```diff
@@ -60,4 +60,28 @@ export abstract class Auth {
 
     return { secretKey, name: accountName };
   }
+
+  async getExtendedCredentials(): Promise<{
```
Comment (serjonya-trili) [inline packages/social-auth/src/Auth.ts:64]: it's better to just extend the original `getCredentials` method. The code is essentially a duplicate. it wouldn't hurt to have more properties, so it's not a breaking change either.

File: packages/social-auth/src/Auth.test.ts [hunk 2]
```diff
@@ -52,50 +60,104 @@ describe("Auth", () => {
...
-          await expect(provider.getCredentials()).resolves.toEqual({
-            secretKey: resultKey,
-            name: typeof resultName === "function" ? resultName(provider) : resultName,
+          it.each([FacebookAuth, TwitterAuth])("handles login for %p", async Provider => {
+            const provider = new Provider(redirectSurface);
+            jest.spyOn(provider, "login").mockImplementation(() =>
+              Promise.resolve({
+                finalKeyData: { privKey: finalKey },
+                oAuthKeyData: { privKey: oAuthKey },
+                userInfo: { email, name, profileImage, verifierId },
```
Comment (serjonya-trili) [inline packages/social-auth/src/Auth.test.ts:147]: did you actually check that all those providers give you such a response? the schema might be different

**Review by serjonya-trili — CHANGES_REQUESTED** [review]

File: packages/social-auth/src/Auth.test.ts (hunk 2 above)
Comment (asiia-trilitech) [reply packages/social-auth/src/Auth.test.ts:147]: Yep, checked for all providers we use, even if email is undefined, the field is still there. And the rest of the used fields are present

File: packages/social-auth/src/Auth.ts (hunk 1 above)
Comment (asiia-trilitech) [reply packages/social-auth/src/Auth.ts:64]: Good point. 

The only difference is account name - is that alright?

getCredentials: `const accountName = userInfo.email || userInfo.name || this.idpName;`

getExtendedCredentials: `name: userInfo.name`

Comment (serjonya-trili) [reply packages/social-auth/src/Auth.ts:64]: that's fine. just update the code that uses this bit so that this logic is preserved

Comment (asiia-trilitech) [reply packages/social-auth/src/Auth.ts]: All done now. 
PTAL at https://github.com/trilitech/umami-v2/pull/1626/commits/f86a3b2a03bb60d6753a9b9d6f74d606cb59af26

Not related to this PR, but useOnboardWithSocial.tsx seems identical for desktop & web, are we planning on moving them to packages and reusing?

**Comment by asiia-trilitech** [comment]
@serjonya-trili could you please check this PR as well, I need it for this change

https://github.com/trilitech/umami-embed-npm/pull/10

File: packages/social-auth/src/Auth.ts (hunk 1 above)
Comment (serjonya-trili) [reply packages/social-auth/src/Auth.ts]: yes, we do have such plans

File: apps/desktop/src/components/Onboarding/connectOrCreate/useOnboardWithSocial.tsx [hunk 3]
```diff
@@ -24,13 +24,13 @@ export const useOnboardWithSocial = (idp: Auth.IDP, onAuth: () => void) => {
...
         async () => {
-          const { secretKey, name } = await withTimeout(
+          const { secretKey, name, id } = await withTimeout(
             () => Auth.forIDP(idp, "desktop").getCredentials(),
             LOGIN_TIMEOUT
           );
           const { pk, pkh } = await getPublicKeyPairFromSk(secretKey);
-          restoreSocial(pk, pkh, name, idp);
-          toast({ description: `Successfully added ${name} account`, status: "success" });
+          restoreSocial(pk, pkh, name || id, idp);
```
Comment (serjonya-trili) [inline apps/desktop/src/components/Onboarding/connectOrCreate/useOnboardWithSocial.tsx:32]: here it should be `email || name || id` to preserve [the same logic](https://github.com/trilitech/umami-v2/pull/1626/files#diff-159e8554a760382f6ca9a2bcb9f057d168ccb433f37c7c2efaee615564fd7289L59)

**Review by serjonya-trili — CHANGES_REQUESTED** [review]

File: apps/desktop/src/components/Onboarding/connectOrCreate/useOnboardWithSocial.tsx (hunk 3 above)
Comment (serjonya-trili) [reply apps/desktop/src/components/Onboarding/connectOrCreate/useOnboardWithSocial.tsx:32]: this please
> The other option could be to keep returning const userData = { pk, pkh, typeOfLogin: loginType, id, name, emailAddress: email, imageUrl }; to dapps that use umami embed
But on all umami surfaces still use email || name || id as a name (e.g. desktop, embed modals)

Comment (asiia-trilitech) [reply apps/desktop/src/components/Onboarding/connectOrCreate/useOnboardWithSocial.tsx]: Fixed now, thank you!

**Pull request merged**